js/private/node-patches/src/fs.ts

```diff
@@ -613,13 +617,55 @@ export const patcher = (fs: any = _fs, roots: string[]) => {
         oneHop(loc, cb)
     }
 
+    const hopLinkCache = new Map<string, string | typeof HOP_NON_LINK>()
+    function readHopLink(p: string) {
```

Comment:
This caches all hop-related metadata that can be computed from just the path `p` off the filesystem, it does not take things like `nested` or `escapedHop` from the original method into account.

Reply:
call it `readHopLinkSync` as per fs convention?

Reply:
Done (30s before I read this)